services/can_logger/can_logger.py

```diff
@@ -59,93 +74,220 @@ def db_init():
+
+        # Sometimes things are not setup like we expect. Live and let live
+        if not os.path.isfile(path):
+            print(f'{network} does not have a type file. Skipping')
+            continue
+
+        # Open the file and read it
+        with open(path) as typefile:
+            networktype = typefile.read().strip()
+
+        # 280 is the type for CAN. 'Documentation' here:
+        # https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/if_arp.h#L56
+        if networktype.isdigit() and int(networktype) == 280:
+            print('\t', network, ' appears to be a CAN link')
+            can_interfaces.append(network)
+
+    if len(can_interfaces) <= 0:
+        print('FATAL: No CAN interfaces found')
+        sys.exit(-1)
+
+    print(len(can_interfaces), ' found: ', can_interfaces)
+
+    return can_interfaces
+
+
+def log_can(can_interface):
+
+    print(f'Logging {can_interface}')
+    
+    frame = ''
+    rx_buff = []
+    wr_buff = []
+    socket_connected = False
+    buff = collections.deque(maxlen=3)
+    
+    # Initialize socket
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+    # Connect to socketcand
+
+    while(socket_connected is False):
+
+        try:
+
+            s.connect((host_ip, int(host_port)))
+
+        except(ConnectionRefusedError):
+
+            print('Could not connect to socketcand. Connection Refused. \
+                    Retrying...')
+            time.sleep(10)
+            socket_connected = False
+
+        else:
+
+            print(f'Successfully connected to socketcand at',
+                  '{host_ip}: {host_port}')
+            socket_connected = True
+
+    # Receive socketcand's response
+
+    s.recv(32)
+
+    # Connect to exposed CAN interface
+
+    s.sendall(b'< open ' + can_interface.encode('utf-8') + b' >')
+
+    s.recv(32)
+
+    # Set socket to 'rawmode' to receive every frame on the bus.
+    s.sendall(b'< rawmode >')
+
+    s.recv(32)
+
+    # Receive frames in a 128-byte buffer and then decode the bytes object.
+    # Then strip some characters to clean up the received frame and then split
+    # the resulting string to get the timestamp, CAN ID and CAN frame.
+
+    while(True):
+
+        socket_buff = s.recv(54)
+        buff.append(socket_buff)
+        
+        frame_buff = list(buff)
+        
+        if(len(frame_buff) > 2):
+           
+            frame = frame_buff[1] + frame_buff[2]
+            frame = frame.decode("utf-8").split("<")
+            frame = frame[1].strip('>').split(' ')
+
+        try:
+
+            (timestamp, can_bus, can_id, can_data) = (frame[3], can_interface,
+                                                      frame[2], frame[4])
+
+        except(IndexError):
+
+            print(f'Error logging CAN frame at {can_interface}. Skipping...')
+
+        else:
+
+            timestamp = datetime.fromtimestamp(float(timestamp)).isoformat()
+            rx_buff.append((timestamp, can_bus, can_id, can_data))
```

Comment:
The comment above does a good job of explaining what the purpose of this section is, but I would love a few more in here explaining what `socket_buff`, `frame_buff`, `buff`, `frame`, etc contain and the transformations you are doing to each. I get that you are extracting can data out it but how?